### ISSUE TITLE
Building 

### ISSUE THREAD

**Issue by amdlopes** [emgucv/emgucv#108]
Setup:

xcode 9.2
updated most recent visual studio, xamarin packages
mac os x sierra 10.12.16

Building from source EMGUCV_3_1_R16_12 using instructions. (CMAKE NO FFMPEG because -lavcodec gives linking problems whilst building opencv)

error:
../emgucv/Emgu.CV.Extern/core/core_cuda_c.cpp:41:17: error: ordered comparison between pointer and zero ('int *' and 'int')
   if (deviceId < 0)
       ~~~~~~~~ ^ ~
1 error generated.
make[2]: *** [Emgu.CV.Extern/CMakeFiles/cvextern.dir/core/core_cuda_c.cpp.o] Error 1
make[1]: *** [Emgu.CV.Extern/CMakeFiles/cvextern.dir/all] Error 2
make: *** [all] Error 2

This happened at 95% building opencv. Frustrating building emgucv.

Dont know what to do. Looking for help. Cant seem to build any tag 3.2, 3.1, ... very frustrating.

**Comment by emgucv** [comment]
The specific bug has been fixed in this commit:

0fde19c8a036f955f36e034a50a73300ed5573e8

It is included in the 3.3 release. 

Older version of XCode, at the release time of the Emgu CV 3.2, 3.1 releases will not generate compilation error before this commit, but the more recent version of XCode (e.g. 9.2) will result in compilation error for the older tags such as 3.2 and 3.1. If you are using Xcode 9.2, please use 3.3 tag or master head.

**Comment by emgucv** [comment]
Closing ticket now.
